compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4184,6 +4186,18 @@ TR::Register *OMR::X86::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::C
       {
       return TR::TreeEvaluator::andORStringEvaluator(node, cg);
       }
+   else if ((symbol->getRecognizedMethod() == TR::com_ibm_crypto_provider_AEScryptInHardware_cbcEncrypt)
+         && cg->enableAESInHardwareTransformations() && !disableECCP256AESCBC
+         && !(gcMode == TR_WrtbarCardMarkIncremental || gcMode == TR_WrtbarRealTime) && TR::Compiler->target.is64Bit())
```

Comment:
As far as I know, this code is there to ensure we do not apply this optimization on arraylets.
The code is added from VM28. It might be a better idea to use a function to check whether it is an arraylet mode.